### PR TITLE
Update branch deletion code

### PR DESCRIPTION
**Pull request by weshaggard**
- Update to use the gh cli for github operations
- Switch to using graphql for pulling branches as the rest api starts to fail after a certain number of branches and we can also pull back all the associated data like pull request info and commit date all in one query.

Fixes https://github.com/Azure/azure-sdk-tools/issues/5091

### CODE REVIEW

File: eng/common/scripts/Delete-RemoteBranches.ps1 [hunk 1]
```diff
@@ -14,126 +12,121 @@ param(
...
+    }
+'@
+
+  $all_branches = gh api graphql --paginate -F owner=$owner -F repo=$repo -F refPrefix='refs/heads/' -f query=$query `
+    --jq '.data.repository.refs.nodes[] | { name, commitUrl: .target.commitUrl, committedDate: .target.committedDate, pullRequests: .associatedPullRequests.nodes }' | ConvertFrom-Json
+
+  if ($LASTEXITCODE) {
+    LogError "Failed to retireve branches for '$owner' and '$repo' running query '$query'"
+    exit 1
+  }
```
Comment (benbp) [inline eng/common/scripts/Delete-RemoteBranches.ps1:56]:
```suggestion
  if ($LASTEXITCODE) {
    LogError "Failed to retrieve branches for '$owner' and '$repo' running query '$query'"
    exit $LASTEXITCODE
  }
```

File: eng/common/scripts/Delete-RemoteBranches.ps1 [hunk 2]
```diff
@@ -14,126 +12,121 @@ param(
...
-  try 
-  {
-    if ($PSCmdlet.ShouldProcess("[ $branchName ] in [ $RepoId ]", "Deleting branches on cleanup script")) {
-      Remove-GitHubSourceReferences -RepoId $RepoId -Ref $branch -AuthToken $AuthToken
-      Write-Host "The branch [ $branchName ] with sha [ $($res.object.sha) ] in [ $RepoId ] has been deleted."
+  $commitUrl = $branch.commitUrl
+  if ($PSCmdlet.ShouldProcess("'$branchName' in '$RepoId'", "Deleting branch on cleanup script")) {
+    gh api "repos/${RepoId}/git/refs/heads/${branchName}" -X DELETE
+    if ($LASTEXITCODE) {
+      LogError "Delection of branch '$branchName` failed"
```
Comment (benbp) [inline eng/common/scripts/Delete-RemoteBranches.ps1:128]:
```suggestion
      LogError "Deletion of branch '$branchName` failed"
```

File: eng/pipelines/branch-cleanup.yml [hunk 3]
```diff
@@ -16,6 +16,16 @@ parameters:
     - Azure/azure-sdk-for-net
     - Azure/azure-sdk-for-python
 
+- name: ForkAutomationBranches
+  type: object
+  default:
+    - azure-sdk/azure-powershell
```
Comment (benbp) [inline eng/pipelines/branch-cleanup.yml:22]: Are we responsible for this repo?

**Review by benbp — APPROVED** [review]

File: eng/pipelines/branch-cleanup.yml (hunk 3 above)
Comment (weshaggard) [reply eng/pipelines/branch-cleanup.yml:22]: It is part of the sdk automation coming from the specs repo and it uses a fork in our bot account so I feel like we should have it in the list.

**Pull request merged**